### DISCUSSION TITLE
Android Private DNS vs in-app DoH vs in-app DNS Proxy

### DISCUSSION THREAD

**Discussion by das22** [celzero/rethink-app#934] [Q&A] [answered]
Hi,

Firstly, many thanks for the phenomenal app! 
I've been using it on my Android phone for a while together with NextDNS and I was wondering what a recommended configuration could be assuming combining the two is not a bad idea for some reason. The way I see it, I can set NextDNS within Android's "Private DNS" settings and set the app to System DNS. Alternatively, I can leave Android settings unchanged and set DoH or DNS Proxy to use NextDNS. Are there any "best" solutions among the 3? For example, would all of the all of the above offer a similar level of privacy or do any of these allow phone apps to bypass the DNS?

What struck me immediately during experimentation is that the app logs show many different DNS names when I use the in-app DoH whereas when using there system's Private DNS, the log is dominated by *nextdns.io entries. Is this indicative that not everything is using NextDNS in the former case?

Apologies if there was a topic on this already. I tried searching but couldn't find one. 

Cheers!

**Comment by ignoramous** [comment, marked as answer]
Hi,

> Are there any "best" solutions among the 3? For example, would all of the all of the above offer a similar level of privacy or do any of these allow phone apps to bypass the DNS?

I'd you intend to use *Private DNS*, the Rethink app is of little value, unless you want its Firewall to monitor IP traffic. Otherwise, the better setup is to use Rethink's DoH (with nextdns if you prefer), since Rethink can then match domain names to IPs for outgoing connections.

> ...whereas when using there system's Private DNS, the log is dominated by *nextdns.io entries

That's because *Private DNS* overrides Rethink. All DNS queries are sent to Rethink encrypted.

> ...this indicative that not everything is using NextDNS in the former case?

No. It indicates that the Rethink app isn't incharge of DNS anymore.

Let me know if these answers weren't clear...

**Reply by das22** [reply]
Crystal clear - thank you!
